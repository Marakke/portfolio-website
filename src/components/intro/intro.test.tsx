import React from "react";
import { render } from "@testing-library/react";
import * as Gatsby from "gatsby";
import Intro from "./index";

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`);

const mockUseStaticQuery = {
  intro: {
    title: "Test Title",
    subtitle: "Test Subtitle",
    descriptionTitle: "Test Description Title",
    descriptionBody: "Test Description Body",
    descriptionFooter: "Test Description Footer",
    image: {
      fluid() {
        return {
          aspectRatio: 1,
          src: "mock/image/path.jpg",
          srcSet: "mock/srcSet",
          sizes: "(max-width: 800px) 100vw, 800px",
        };
      },
      alt: "Test Alt Text"
    }
  }
};

describe("Intro", () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("renders the Intro component correctly", () => {
    const { container, getByTestId, getByAltText, getByRole } = render(<Intro />);
    expect(container).toMatchSnapshot();

    const introContainer = getByTestId("intro-container");
    expect(introContainer).toBeInTheDocument();

    const titleElement = getByRole("heading", { name: "Test Title - Test Subtitle" });
    expect(titleElement).toBeInTheDocument();

    const imageElement = getByAltText("Test Alt Text");
    expect(imageElement).toBeInTheDocument();

    const paragraphElements = container.querySelectorAll("p");
    expect(paragraphElements.length).toEqual(8);
    expect(paragraphElements[0].textContent).toBe("Test Description Title");
  });
});
