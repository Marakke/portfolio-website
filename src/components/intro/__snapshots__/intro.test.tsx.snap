// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders the Intro component correctly 1`] = `
<div>
  <div
    class="intro-container"
    data-testid="intro-container"
    id="introduction"
  >
    <h2>
      Test Title
       - 
      <span
        class="intro-subtitle"
      >
        Test Subtitle
      </span>
    </h2>
    <div
      class="intro-content"
    >
      <div
        class="image gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          aria-hidden="true"
          style="width: 100%;"
        />
        <picture>
          <img
            alt="Test Alt Text"
            loading="lazy"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcset="undefined" /&gt;&lt;img loading="lazy" src="" alt="Test Alt Text" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <div>
        <p
          class="description"
        >
          Test Description Title
        </p>
        <p
          class="description"
        >
          Test Description Body
        </p>
        <p
          class="description"
        >
          Test Description Footer
        </p>
      </div>
    </div>
    <div
      class="lines"
    >
      <p
        class="line"
      />
      <p
        class="line"
      />
      <p
        class="line"
      />
      <p
        class="line"
      />
      <p
        class="line"
      />
    </div>
  </div>
</div>
`;
